source/features/preserve-text-messages.js

```diff
@@ -0,0 +1,116 @@
+import debounce from 'lodash.debounce';
+import {observeEl} from '../libs/utils';
+
+let DMDialogOpen = false;
+
+function onDMDialogOpen() {
+	observeEl('#dm_dialog', async mutations => {
+		for (const mutation of mutations) {
+			if (mutation.target.style.display === 'none') {
+				DMDialogOpen = false;
+			} else {
+				DMDialogOpen = true;
+				onDMOpen();
+			}
+		}
+
+		if (!DMDialogOpen) {
+			const savedMessages = await browser.storage.local.get();
+			const idsOfMessagesToRemove = [];
+
+			for (const id in savedMessages) {
+				if (isEmptyMsgInput(savedMessages[id])) {
+					idsOfMessagesToRemove.push(id);
+				}
+			}
+
+			await browser.storage.local.remove(idsOfMessagesToRemove);
+		}
+	}, {attributes: true});
+}
+
+function onDMOpen() {
+	const conversationOptions = {
+		attributes: true,
+		attributeFilter: ['class']
+	};
+
+	observeEl('.DMConversation', mutations => {
+		let conversationId;
+		let messageContainer;
+
+		for (const mutation of mutations) {
+			if (mutation.target.classList.contains('DMActivity--open')) {
+				conversationId = getConversationId();
+				messageContainer = $('#tweet-box-dm-conversation');
+				if (conversationId) {
+					onMessageChange(messageContainer);
+				}
+			}
+			if (conversationId) {
```

Comment:
Do you need to run this after running `onMessageChange` above? If not, maybe `return` on line `48`, before the closing `}`